app/src/main/java/org/wikipedia/descriptions/DescriptionEditFragment.kt

```diff
@@ -0,0 +1,442 @@
+            }
+        }
+
+        private fun getEditTokenThenSave(forceLogin: Boolean) {
+            if (csrfClient == null) {
+                return
+            }
+            csrfClient?.request(forceLogin, object : CsrfTokenClient.Callback {
+                override fun success(token: String) {
+                    if (shouldWriteToLocalWiki()) {
+                        // If the description is being applied to an article on English Wikipedia, it
+                        // should be written directly to the article instead of Wikidata.
+                        postDescriptionToArticle(token)
+                    } else {
+                        postDescriptionToWikidata(token)
+                    }
+                }
+
+                override fun failure(caught: Throwable) {
+                    editFailed(caught, false)
+                }
+
+                override fun twoFactorPrompt() {
+                    editFailed(LoginFailedException(resources
+                            .getString(R.string.login_2fa_other_workflow_error_msg)), false)
+                }
+            })
+        }
+
+        private fun postDescriptionToArticle(editToken: String) {
+            val wikiSite = WikiSite.forLanguageCode(pageTitle!!.wikiSite.languageCode())
+            disposables.add(ServiceFactory.get(wikiSite).getWikiTextForSection(pageTitle!!.prefixedText, 0)
+                    .subscribeOn(Schedulers.io())
+                    .flatMap { mwQueryResponse ->
+                        var text = mwQueryResponse.query()!!.firstPage()!!.revisions()[0].content()
+                        val baseRevId = mwQueryResponse.query()!!.firstPage()!!.revisions()[0].revId
+                        text = updateDescriptionInArticle(text, binding.fragmentDescriptionEditView.description)
+                        ServiceFactory.get(wikiSite).postEditSubmit(pageTitle!!.prefixedText, "0", null,
+                                when (action) {
+                                    DescriptionEditActivity.Action.ADD_DESCRIPTION -> SuggestedEditsFunnel.SUGGESTED_EDITS_ADD_COMMENT
+                                    DescriptionEditActivity.Action.TRANSLATE_DESCRIPTION -> SuggestedEditsFunnel.SUGGESTED_EDITS_TRANSLATE_COMMENT
+                                    else -> ""
+                                },
+                                if (AccountUtil.isLoggedIn) "user"
+                                else null, text, null, baseRevId, editToken, null, null)
+                                .subscribeOn(Schedulers.io())
+                    }
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe({ result ->
+                        if (result.hasEditResult() && result.edit() != null) {
+                            when {
+                                result.edit()!!.editSucceeded() -> {
+                                    Handler().postDelayed(successRunnable, TimeUnit.SECONDS.toMillis(4))
```

Comment:
since `Handler()` is deprecated, let's switch to using `requireView.postDelayed()`.